src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -166,11 +168,20 @@ private void declareNamespace(String namespace, String provide, TypedVar symbol,
       new TreeWalker(symbol, namespace).walk(true);
     } else {
       // JSCompiler treats "foo.x" as one variable name, so collect all provides that start with
-      // $provide + ".".
-      String prefix = provide + ".";
+      // $provide + "." but are not sub-properties.
+      Set<String> desiredSymbols = new HashSet<>();
+
+      TypedVar o = Iterables.find(allSymbols, nameIs(provide));
+      if (o.getType().isObject()) {
+        ObjectType objType = (ObjectType) o.getType();
+        for (String property : objType.getPropertyNames()) {
+          desiredSymbols.add(o.getName() + "." + property);
+        }
+      }
+
       for (TypedVar other : allSymbols) {
```

Comment:
because the order is different.
I'm not sure it's that important for the order to match the declarations in the js file, but it seems desirable compared with any other ordering.